### DISCUSSION TITLE
Regarding pleomorphic enveloped viruses

### DISCUSSION THREAD

**Discussion by Ankushcreatedroid** [nextpyp/.github#159] [Q&A] [answered]
Can anybody forum suggest a good work flow to do uniform spike picking over a segemented pleomorphic virus geometry. Also if some body can give idea on how to segment pleomorphic virus to get there geometries out for picking uniform spikes over the membrane surface ?

Thank you
Ankush Roy

**Comment by abartesaghi** [comment]
For segmenting virions (spherical or pleomorphic), look at the [Tomogram segmentation](https://nextpyp.app/files/pyp/latest/docs/guide/segmentation.html) section in the user guide.

Once you have successfully segmented your virions, follow these steps to pick particles: 
1) create a _Particle picking (open surfaces, filaments)_ block (downstream of the segmentation block)
2) select the size of your particles (this is used for display purposes only)
3) choose between `template search` or `uniform` (and set the required parameters for the selected method)

This will allow you to run template search or pick uniform positions constrained to the segmented surfaces.

**Reply by Ankushcreatedroid** [reply]
thank you so much for reply one more thing i wanted to ask is there any way to import already recontructed tomograms to nextPYP and do picking and followed by exporting to relion5??

**Reply by abartesaghi** [reply]
You cannot directly import tomograms into nextPYP, but you can import existing Relion projects (compatible with both versions 4 and 5). From there, you can pick your particles in nextPYP and then export them back to Relion.

**Reply by Ankushcreatedroid** [reply]
Thank you so much for the reply. I am sorry for such late response

**Reply by Ankushcreatedroid** [reply]
Just general query on cryoCARE denoising my use case is that i have trained a model on 5 Tilt-series. Now i have a batch of total 300 TS can i use that model from 5 titl-series training to denoise rest of tilt-series batch.   I tried this but after doing (denoise eval) using the model from small dataset from filtering and applying to large tilt-series batch the output doesn't give denoised tomograms it is just re-outputing same tomograms non-denoised.

**Reply by abartesaghi** [reply]
We would need more information to troubleshoot this issue. We recently added an option that lets you switch between the original and denoised versions of each tomogram, what version of nextPYP are you using?

**Reply by Ankushcreatedroid** [reply]
i am currently updated nextPYP 0.7.3.  Thank you will be waiting for a some sort of update before even running that big chunk of data for single denoise train.

**Reply by Ankushcreatedroid** [reply]
<img width="1495" height="710" alt="Screenshot From 2026-02-17 23-56-32" src="https://github.com/user-attachments/assets/7e2a71f7-050d-4f17-a186-55df3e8ad5f3" />
<img width="1238" height="731" alt="Screenshot From 2026-02-17 23-56-01" src="https://github.com/user-attachments/assets/c576ddb7-0417-4581-bc0f-37b847df7d7c" />
so u can see that i have succesfully trained model for 5 TS from batch of 270 TS but when i apply that trained model to whole batch then each position_*.mrc tomogram complains this message that in .hd5 file it can't find the file corresponding to the position.

**Reply by abartesaghi** [reply, marked as answer]
As I mentioned, we would need some additional details to properly troubleshoot this. Also, there’s no need to run cryoCARE denoising on the entire dataset right away. You could first select a few tilt-series as a test subset. Once you confirm that the results are as expected, you can proceed with processing the full dataset. This can be done easily in nextPYP using tilt-series filters. By the way, what settings are you using for training and what does the plot of the loss function look like for your training block?

**Reply by Ankushcreatedroid** [reply]
As you mentioned and also in tutorial i filtered some 5 TS from whole batch for denoising in this session. I got my expected result for surface protein to be more clear. Though my loss function oscillates around 6 as you see in screenshot. For the parameter i used particle box size of 72 default because i have arrays of particles on membrane surface plus training epochs of 100 and steps 200 as suggested in original cryoCARE website tutorial. 
<img width="1259" height="534" alt="Screenshot From 2026-02-18 00-07-57" src="https://github.com/user-attachments/assets/503cd2c9-476f-41c5-9913-49a39de811f8" />
<img width="672" height="530" alt="Screenshot From 2026-02-18 00-07-22" src="https://github.com/user-attachments/assets/64a835c2-9b53-4af8-9f3d-f0aa8204c6f1" />

**Reply by Ankushcreatedroid** [reply]
To the previous suggestion you have given. I followed it and got good enough membrane segmentations of virus but now when i am doing uniform picking over segmented membrane why uniform picking is happening both inside and outside membrane and though i want only normals facing outwards. Any specific reason. 
<img width="229" height="222" alt="Screenshot From 2026-02-19 23-02-01" src="https://github.com/user-attachments/assets/8fac3815-03cc-4f01-be77-15e25fe21c1c" />
<img width="229" height="222" alt="Screenshot From 2026-02-19 23-01-24" src="https://github.com/user-attachments/assets/4d7653cc-685e-47aa-8d08-91a2c755777b" />
<img width="229" height="222" alt="Screenshot From 2026-02-19 23-01-15" src="https://github.com/user-attachments/assets/57ff4f27-5e03-4e77-a1d3-2184f3880b3b" />

**Reply by Ankushcreatedroid** [reply]
This are my uniform picking parameters. 
<img width="1284" height="307" alt="Screenshot From 2026-02-19 23-06-37" src="https://github.com/user-attachments/assets/4a20fd19-798d-4d19-947c-16dfa872e52b" />

**Reply by abartesaghi** [reply]
What you are seeing is, unfortunately, the expected behavior. Because the membranes generated by MemBrain-seg are several voxels thick, the "outside" region of the membrane includes both an internal and an external surface. Each of these surfaces produces uniformly spaced positions. 

That said, we do have newer tools that allow you to select only the outer surface. These tools have not yet been released, but I would be happy to share instructions if you email me directly.

**Reply by abartesaghi** [reply]
> As you mentioned and also in tutorial i filtered some 5 TS from whole batch for denoising in this session. I got my expected result for surface protein to be more clear. Though my loss function oscillates around 6 as you see in screenshot. For the parameter i used particle box size of 72 default because i have arrays of particles on membrane surface plus training epochs of 100 and steps 200 as suggested in original cryoCARE website tutorial. <img alt="Screenshot From 2026-02-18 00-07-57" width="1259" height="534" src="https://private-user-images.githubusercontent.com/188021757/551132699-503cd2c9-476f-41c5-9913-49a39de811f8.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzE5NTEzOTMsIm5iZiI6MTc3MTk1MTA5MywicGF0aCI6Ii8xODgwMjE3NTcvNTUxMTMyNjk5LTUwM2NkMmM5LTQ3NmYtNDFjNS05OTEzLTQ5YTM5ZGU4MTFmOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMjI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDIyNFQxNjM4MTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YjYwOWRkMzFlMjkxMDdmYzYxMGVkNWM2MzZhMjk0Y2IyOWU1NWRhZGZjM2FlNzk3OGNkNjBjYmNkMGNhYzZiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.pl7bhQ1gcvMDoU_qeRTXWa8w0SS7ghsCJM7DBA_IBvc"> <img alt="Screenshot From 2026-02-18 00-07-22" width="672" height="530" src="https://private-user-images.githubusercontent.com/188021757/551132714-64a835c2-9b53-4af8-9f3d-f0aa8204c6f1.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzE5NTEzOTMsIm5iZiI6MTc3MTk1MTA5MywicGF0aCI6Ii8xODgwMjE3NTcvNTUxMTMyNzE0LTY0YTgzNWMyLTliNTMtNGFmOC05ZjNkLWYwYWE4MjA0YzZmMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMjI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDIyNFQxNjM4MTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03YmEwODE5YTQwOTQwMjc1MDE5NzBhODEyMTdmYWVlODA4YWI1MDU5NTFhNTUyN2Q1MDkzNWIyNmI3MmJmOTdkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.RbXPwpY-yRJWrJAQ1iRy1MSFPO9BAuq6hwtc9JS9_j8">

The values for the loss function you are getting are quite high (we typically see values lower than 1). Also, we were unable to run cryoCARE with the parameters shown in the screenshot since apparently the number of **Sub-tomograms for statistics** is too large. What are the dimensions of the tomograms you are using for training?

**Reply by Ankushcreatedroid** [reply]
Thank you for your response in the mean time i read about Poisson distribution based normals drawn on surfaces and got to know about patchy membrain-seg results and due to which i understand the behaviour also uniform picking is giving downstream to the membrain-seg. I will respond to you through mail for your instructions. Again thank you for your response Dr Alberto.

**Reply by Ankushcreatedroid** [reply]
Secondly i didn't know what error rate to expect from the cryoCARE denoising protocol and also i am working with pretty much densly packed surface glycoprotein of enveloped virus so assumed to have a large number of subtomograms. Later i found a membrain-seg model for non-denoised tomograms which worked awesome for my case . So i removed the denoising part from my workflow. My current tomogram size is bin8 512*512*375 (X*Y*Z).
